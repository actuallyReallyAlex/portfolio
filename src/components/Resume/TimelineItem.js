import React from 'react'
import PropTypes from 'prop-types'
import { Box, Heading, Paragraph, Text } from 'grommet'
import * as icons from 'grommet-icons'

const TimelineItem = ({
  endDate,
  icon,
  organization,
  present,
  size,
  startDate,
  summary,
  title
}) => {
  const Icon = icons[icon]

  return (
    <Box
      align="start"
      direction={size === 'small' ? 'column' : 'row'}
      fill="horizontal"
      gap={size === 'small' ? 'medium' : 'xlarge'}
      justify="center"
      margin={size === 'small' ? { vertical: 'medium' } : undefined}
    >
      <Box
        style={{ textAlign: `${size === 'small' ? 'center' : 'right'}` }}
        width={size === 'small' ? '100%' : '30%'}
      >
        <Heading level="3" margin={size === 'small' ? 'none' : undefined}>
          {title}
        </Heading>
        <Text>
          {startDate} {startDate && '-'} {present ? 'Present' : endDate}
        </Text>
      </Box>
      <Box
        align="center"
        pad="medium"
        width={size === 'small' ? '100%' : '10%'}
      >
        <Icon />
      </Box>
      <Box align={size === 'small' ? 'center' : undefined} width="medium">
        <Heading level="4" margin="none">
          {organization}
        </Heading>
        <Box
          background="neutral-3"
          height="3px"
          margin={{ top: 'small', bottom: 'medium' }}
          round="small"
        />
        <Paragraph textAlign={size === 'small' ? 'center' : undefined}>
          {summary}
        </Paragraph>
      </Box>
    </Box>
  )
}

TimelineItem.propTypes = {
  endDate: PropTypes.string,
  icon: PropTypes.string.isRequired,
  organization: PropTypes.string.isRequired,
  present: PropTypes.bool,
  size: PropTypes.string.isRequired,
  startDate: PropTypes.string,
  summary: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
}

export default TimelineItem
