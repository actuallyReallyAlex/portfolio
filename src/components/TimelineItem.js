import React from 'react'
import PropTypes from 'prop-types'
import { Box } from 'grommet'
import TimelineIcon from './TimelineIcon'
import TimelineHeading from './TimelineHeading'
import TimelineContent from './TimelineContent'

const TimelineItem = props => {
  return (
    <Box direction="row" pad="medium" margin="none">
      <Box basis="medium" textAlign="right">
        {props.present ? (
          <TimelineHeading
            title={props.title}
            startDate={props.startDate}
            present
          />
        ) : (
          <TimelineHeading
            title={props.title}
            startDate={props.startDate}
            endDate={props.endDate}
          />
        )}
      </Box>
      <Box pad="medium" basis="small" align="center">
        <TimelineIcon icon={props.icon} />
      </Box>
      <Box basis="large">
        <TimelineContent subtitle={props.subtitle} summary={props.summary} />
      </Box>
    </Box>
  )
}

TimelineItem.defaultProps = {
  endDate: 'End Date',
  icon: 'briefcase',
  present: false,
  startDate: 'Start Date',
  subtitle: 'Example Subtitle',
  summary: 'Example Summary',
  title: 'Example Title'
}

TimelineItem.propTypes = {
  /** End date for item. */
  endDate: PropTypes.string,
  /** Icon to be displayed in the middle of the item. */
  icon: PropTypes.string,
  /** If this item is still ongoing, apply the `present` prop. */
  present: PropTypes.bool,
  /** Start date for item. */
  startDate: PropTypes.string.isRequired,
  /** A smaller title with more description. */
  subtitle: PropTypes.string.isRequired,
  /** A summary of the item. */
  summary: PropTypes.string.isRequired,
  /** A title for the item. */
  title: PropTypes.string.isRequired
}

export default TimelineItem
