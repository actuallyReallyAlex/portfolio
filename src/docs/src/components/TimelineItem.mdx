---
name: TimelineItem
route: /components/TimelineItem
menu: Components
---

import { Playground, PropsTable } from 'docz'
import TimelineItem from '../../../components/TimelineItem'

# TimelineItem

An item to be displayed among other items in a timeline.

## Example

---

<TimelineItem />

---

## Properties

<PropsTable of={TimelineItem} />

## Basic Usage

### Event

<Playground>
  <TimelineItem 
    startDate="March 24th, 2005" 
    endDate="March 25th, 2005" 
    title="Stapler in Jell-O" 
    subtitle="Jim put my stapler in Jell-O." 
    summary="I really can't believe he did this. I can't stand to sit this close to him every day."
  />
</Playground>

### Job

<Playground>
  <TimelineItem
    title="HR Specialist"
    startDate="June 2016"
    present
    icon="analytics"
    subtitle="Quantum Learning Network"
    summary="Responsible for staffing the SuperCamp summer program - recruitment, interviewing, contracting over 100 new employees. Created a staff website, designed HTML newsletters, and created numerous media campaigns as a key member of the SuperCamp Programs Department."
  />
</Playground>

### Education

<Playground>
  <TimelineItem
    title="Bachelor of Science"
    startDate=""
    present
    icon="book"
    subtitle="Portland State University"
    summary="Currently pursuing a Bachelor of Science in Biology."
  />
</Playground>