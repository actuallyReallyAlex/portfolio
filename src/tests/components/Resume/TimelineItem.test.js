import React from 'react'
import TimelineItem from '../../../components/Resume/TimelineItem'
import { shallow } from 'enzyme'

const defaultProps = {
  icon: 'Code',
  organization: 'Organization',
  summary: 'Summary',
  title: 'Title',
  startDate: 'December 1999',
  endDate: 'January 2000',
  size: 'medium'
}

describe('<PortfoliTimelineItemoItem />', () => {
  test('Should render the <TimelineItem />.', () => {
    const component = shallow(<TimelineItem {...defaultProps} />)
    expect(component).toMatchSnapshot()
  })

  test('Should render the <TimelineItem /> with present prop.', () => {
    const component = shallow(<TimelineItem {...defaultProps} present={true} />)
    expect(component).toMatchSnapshot()
  })
})
