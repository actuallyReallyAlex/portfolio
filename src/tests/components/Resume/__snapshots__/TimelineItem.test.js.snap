// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PortfoliTimelineItemoItem /> Should render the <TimelineItem /> with present prop. 1`] = `ShallowWrapper {}`;

exports[`<PortfoliTimelineItemoItem /> Should render the <TimelineItem />. 1`] = `ShallowWrapper {}`;
